src/logic/safe/transactions/txHistory.ts

```diff
@@ -63,24 +66,24 @@ export const saveTxToHistory = async ({
   signature,
   to,
   valueInWei,
-}: SaveTxToHistoryArgs): Promise<TransactionDetails> => {
+}: SaveTxToHistoryProps): Promise<TransactionDetails> => {
   const address = checksumAddress(safeInstance.options.address)
-  const body = await calculateBodyFrom(
+  const body = await calculateBodyFrom({
     safeInstance,
     to,
-    valueInWei,
+    value: valueInWei,
     data,
     operation,
-    nonce,
+    nonce: nonce.toString(),
```

Comment:
Why have you changed this to a string?

Reply:
Because of the `MultisigTransactionRequest` types. It expects `nonce` as a string.

if I leave it as `number`...
> The expected type comes from property 'nonce' which is declared here on type 'ProposeTxBodyTypes'

Reply:
I get that you're adhering to the types but you added the types yourself (`ProposeTxBodyProps`) in this PR. What's to say it was not expecting a number previously? Have you tested this?

Reply:
I've tested it.
I made the return type in `calculateBodyFrom` to be `MultisigTransactionRequest` because it is the type used when you pass `body` to
<img width="842" alt="Screen Shot 2021-12-17 at 14 33 40" src="https://user-images.githubusercontent.com/32431609/146552481-6c247c73-cce2-4979-aa35-27ebeb22a91b.png">
 
 so I believe it is the correct type
 I believe